source/auth/Credentials.cpp

```diff
@@ -28,6 +28,16 @@ namespace Aws
                 }
             }
 
+            Credentials::Credentials(const aws_credentials *credentials, Allocator *allocator) noexcept
```

Comment:
What is the need for this constructor?

Reply:
The get credentials callback from imds client provides a const aws_credentials parameter. This is used to construct a Credentials object from a const raw aws_credentials.

Reply:
Couldn't the existing constructor have been relaxed?

Reply:
Another possibility (gathering feedback) is changing acquire/release to take const * and then updating everything here and in auth to be appropriately const *.